WIP Leanback

razzeee opened the pull request:
@koying

LS80 committed:
[addons] Preserve strings which are not string ids in enum lvalues

FernetMenta committed:
VideoPlayer: continue normal playback if seek fails

FernetMenta committed:
VideoPlayer: update selectionstreams when stream parameters changes

Paxxi committed:
Change archive to use a fixed length type for string and array lengths to avoid crazy memory allocations.
Also add some rudimentary boundary checking to avoid crashing on corrupt archives

Paxxi committed:
some cleanup of CArchive while touching it

koying committed:
FIX: ignore options in URIUtils::Split

razzeee committed:
[android] leanback recommendations & search

razzeee committed:
[droid] cleanup XBMCApp

Tolriq committed:
Fix screensaver issue with JSON and pictures

Opening a picture or a slideshow from JSON should stop any active screensaver.

razzeee committed:
Don't try to download when url is empty

razzeee committed:
Revert "Drop unused method"

This reverts commit 9c24171d46f602b0a569d9aa8875f1726500a8d7.

razzeee committed:
Fix option for movies

koying commented on addons/resource.language.en_gb/resources/strings.po:
```diff
@@ -11038,7 +11038,10 @@ msgctxt "#20072"
+#: xbmc/windows/GUIWindowFileManager.cpp
+msgctxt "#20073"
+msgid "Android content"
+msgstr ""
```
no

koying commented on configure.ac:
```diff
@@ -117,6 +117,7 @@ AC_DEFUN([XB_PUSH_FLAGS], [
 ])
 
 # version can be overridden by setting the following as ENV vars when running configure
+APP_PACKAGE=${APP_PACKAGE-$(${AWK} '/APP_PACKAGE/ {print $2}' version.txt)}
```
no

koying commented on configure.ac:
```diff
@@ -133,6 +134,7 @@ if test "$APP_NAME" != "" && test "$APP_VERSION_MAJOR" != "" && test "$APP_VERSI
     APP_VERSION=${APP_VERSION}-${APP_VERSION_TAG}
   fi
 
+  AC_SUBST(APP_PACKAGE)
```
no

koying commented on configure.ac:
```diff
@@ -2231,9 +2239,11 @@ fi
 eval "APP_PREFIX=${prefix}"
 eval "APP_LIB_DIR=${libdir}/${APP_NAME_LC}"
 eval "APP_INCLUDE_DIR=${includedir}/${APP_NAME_LC}"
+eval "APP_PACKAGE_DIR=${APP_PACKAGE//./\/}"
```
no

koying commented on configure.ac:
```diff
@@ -2231,9 +2239,11 @@ fi
 AC_SUBST(APP_PREFIX)
 AC_SUBST(APP_LIB_DIR)
 AC_SUBST(APP_INCLUDE_DIR)
+AC_SUBST(APP_PACKAGE_DIR)
```
no

koying commented on tools/android/packaging/Makefile.in:
```diff
@@ -7,7 +7,7 @@ OBJS = libcurl.so \
 
 PLATFORM_OBJS =
-EXCLUDED_ADDONS = 
+EXCLUDED_ADDONS =
```
no

koying commented on tools/android/packaging/Makefile.in:
```diff
@@ -26,7 +26,7 @@ endif
 ARMOVERRIDES=XBMC_OVERRIDE_HOST=arm-linux-androideabi XBMC_OVERRIDE_TOOLCHAIN=$(XBMC_ARM_TOOLCHAIN)
 
-#this fixes a android ndk fuckup where the pathes to 
+#this fixes a android ndk fuckup where the pathes to
```
no

koying commented on tools/android/packaging/Makefile.in:
```diff
@@ -102,7 +102,7 @@ python: | xbmc/assets
 
 res:
-	mkdir -p xbmc/res xbmc/res/raw xbmc/res/values images
+	mkdir -p xbmc/gen/@APP_PACKAGE_DIR@ xbmc/res xbmc/res/raw xbmc/res/values images
```
semi-no. You need the gen but not the APP_PACKAGE_DIR
You can hardcode APP_PACKAGE_DIR to org/xbmc/kodi

koying commented on tools/android/packaging/xbmc/AndroidManifest.xml.in:
```diff
@@ -2,6 +2,7 @@
 <!-- BEGIN_INCLUDE(manifest) -->
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.xbmc.@APP_NAME_LC@"
+    android:installLocation="auto"
```
no

koying commented on tools/android/packaging/xbmc/AndroidManifest.xml.in:
```diff
@@ -13,6 +14,7 @@
     <uses-permission android:name="android.permission.GET_TASKS" />
     <uses-permission android:name="android.permission.BLUETOOTH" />
     <uses-permission android:name="android.permission.MODIFY_AUDIO_SETTINGS" />
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
```
no

koying commented on tools/android/packaging/xbmc/src/org/xbmc/kodi/Main.java.in:
```diff
@@ -79,6 +84,12 @@ public class Main extends NativeActivity implements Choreographer.FrameCallback
+    Window window = getWindow();
+    window.addFlags(WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON);
+  }
+
```
no

koying commented on tools/android/packaging/xbmc/src/org/xbmc/kodi/Main.java.in:
```diff
@@ -10,6 +10,8 @@ import android.util.Log;
 import android.view.View;
 import android.view.Surface;
+import android.view.Window;
+import android.view.WindowManager;
```
no

koying commented on tools/android/packaging/xbmc/src/org/xbmc/kodi/Main.java.in:
```diff
@@ -23,11 +25,14 @@ public class Main extends NativeActivity implements Choreographer.FrameCallback
+  private Intent mNewIntent = null;
 
   native void _onNewIntent(Intent intent);
+  native void _onActivityResult(int requestCode, int resultCode, Intent resultData);
```
no

koying commented on tools/android/packaging/xbmc/src/org/xbmc/kodi/Main.java.in:
```diff
@@ -169,6 +179,9 @@ public class Main extends NativeActivity implements Choreographer.FrameCallback
 
+    Window window = getWindow();
+    window.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+
```
no

koying commented on tools/android/packaging/xbmc/src/org/xbmc/kodi/Main.java.in:
```diff
@@ -190,17 +244,34 @@ public class Main extends NativeActivity implements Choreographer.FrameCallback
+  {
+    _onActivityResult(requestCode, resultCode, resultData);
+  }
+
```
no

koying commented on tools/android/packaging/xbmc/src/org/xbmc/kodi/Main.java.in:
```diff
@@ -190,17 +244,34 @@ public class Main extends NativeActivity implements Choreographer.FrameCallback
+          android.os.Process.killProcess(android.os.Process.myPid());
+        } catch (Exception e) {}
+      }
+    }.start();
```
no

koying commented on tools/android/packaging/xbmc/src/org/xbmc/kodi/Splash.java.in:
```diff
@@ -763,7 +835,9 @@ public class Splash extends Activity {
-    if ((mState != DownloadingObb && mState != InError) && mCachingDone && mExternalStorageChecked) {
+    if ((mState != DownloadingObb && mState != InError) && mCachingDone && mExternalStorageChecked
+        && mNetworkReady)
+    {
```
None of Splash.java.in is needed

koying commented on tools/depends/target/android-sources-ics/android-source/system/core/toolbox/sync.c:
```diff
@@ -0,0 +1,7 @@
+{
+	sync();
+	return 0;
+}
```
no

koying commented on version.txt:
```diff
@@ -1,4 +1,5 @@
 APP_NAME Kodi
+APP_PACKAGE org.xbmc.kodi
```
no

koying commented on xbmc/Application.cpp:
```diff
@@ -5150,6 +5155,6 @@ bool CApplication::NotifyActionListeners(const CAction &action) const
       return true;
   }
-  
+
```
None of Application.cpp

koying commented on xbmc/filesystem/AndroidContentDirectory.cpp:
```diff
@@ -0,0 +1,114 @@
+  docCursor.close();
+
+  return true;
+}
```
none of "Android content" stuff

koying commented on xbmc/filesystem/AndroidContentDirectory.h:
```diff
@@ -0,0 +1,40 @@
+  virtual DIR_CACHE_TYPE GetCacheType(const std::string& strPath) const { return DIR_CACHE_NEVER; }
+};
+}
+
```
none of "Android content" stuff

koying commented on xbmc/filesystem/DirectoryFactory.cpp:
```diff
@@ -96,6 +96,7 @@
 #endif
 #if defined(TARGET_ANDROID)
 #include "AndroidAppDirectory.h"
+#include "AndroidContentDirectory.h"
```
none of "Android content" stuff

koying commented on xbmc/filesystem/DirectoryFactory.cpp:
```diff
@@ -164,6 +165,7 @@ IDirectory* CDirectoryFactory::Create(const CURL& url)
 #if defined(TARGET_ANDROID)
   if (url.IsProtocol("androidapp")) return new CAndroidAppDirectory();
+  if (url.IsProtocol("content")) return new CAndroidContentDirectory();
 #endif
```
none of "Android content" stuff

koying commented on xbmc/filesystem/Makefile.in:
```diff
@@ -77,6 +77,7 @@ SRCS += APKDirectory.cpp
 SRCS += APKFile.cpp
 SRCS += AndroidAppFile.cpp
 SRCS += AndroidAppDirectory.cpp
+SRCS += AndroidContentDirectory.cpp
```
none of "Android content" stuff

koying commented on xbmc/platform/android/activity/JNIMainActivity.cpp:
```diff
@@ -47,6 +47,14 @@ void CJNIMainActivity::_onNewIntent(JNIEnv *env, jobject context, jobject intent
+  if (m_appInstance)
+    m_appInstance->onActivityResult(requestCode, resultCode, CJNIIntent(jhobject(resultData)));
+}
+
```
no

koying commented on xbmc/platform/android/activity/JNIMainActivity.cpp:
```diff
@@ -140,3 +148,9 @@ void CJNIMainActivity::unregisterMediaButtonEventReceiver()
+{
+  call_method<void>(m_context,
+                    "screenOn", "()V");
+}
```
no

koying commented on xbmc/platform/android/activity/JNIMainActivity.h:
```diff
@@ -53,7 +55,8 @@ class CJNIMainActivity : public CJNIActivity, public CJNIInputManagerInputDevice
   virtual void onAudioFocusChange(int focusChange)=0;
   virtual void doFrame(int64_t frameTimeNanos)=0;
-};
+};
```
none of the above

koying commented on xbmc/platform/android/activity/XBMCApp.cpp:
```diff
@@ -102,11 +104,13 @@ CJNIWakeLock *CXBMCApp::m_wakeLock = NULL;
 ANativeWindow* CXBMCApp::m_window = NULL;
 int CXBMCApp::m_batteryLevel = 0;
 bool CXBMCApp::m_hasFocus = false;
+bool CXBMCApp::m_isResumed = false;
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.cpp:
```diff
@@ -102,11 +104,13 @@ CJNIWakeLock *CXBMCApp::m_wakeLock = NULL;
 IInputDeviceEventHandler* CXBMCApp::m_inputDeviceEventHandler = nullptr;
 CCriticalSection CXBMCApp::m_applicationsMutex;
 std::vector<androidPackage> CXBMCApp::m_applications;
+std::vector<CActivityResultEvent*> CXBMCApp::m_activityResultEvents;
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.cpp:
```diff
@@ -167,6 +171,7 @@ void CXBMCApp::onResume()
   CJNIIntentFilter intentFilter;
   intentFilter.addAction("android.intent.action.BATTERY_CHANGED");
   intentFilter.addAction("android.intent.action.SCREEN_ON");
+  intentFilter.addAction("android.intent.action.SCREEN_OFF");
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.cpp:
```diff
@@ -184,6 +189,8 @@ void CXBMCApp::onResume()
     m_applications.clear();
   }
+
+  m_isResumed = true;
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.cpp:
```diff
@@ -207,6 +214,7 @@ void CXBMCApp::onPause()
 #endif
 
   EnableWakeLock(false);
+  m_isResumed = false;
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.cpp:
```diff
@@ -222,7 +230,7 @@ void CXBMCApp::onDestroy()
   if (!m_exiting)
   {
-    XBMC_Stop();
+    CApplicationMessenger::GetInstance().PostMsg(TMSG_QUIT);
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.cpp:
```diff
@@ -259,7 +267,6 @@ void CXBMCApp::onCreateWindow(ANativeWindow* window)
   if(!m_firstrun)
   {
     XBMC_SetupDisplay();
-    XBMC_Pause(false);
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.cpp:
```diff
@@ -280,7 +287,6 @@ void CXBMCApp::onDestroyWindow()
   {
     XBMC_DestroyDisplay();
     m_window = NULL;
-    XBMC_Pause(true);
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.cpp:
```diff
@@ -366,11 +372,6 @@ bool CXBMCApp::ReleaseAudioFocus()
-{
-  return m_hasFocus;
-}
-
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.cpp:
```diff
@@ -422,16 +423,6 @@ void CXBMCApp::run()
-{
-  CApplicationMessenger::GetInstance().PostMsg(TMSG_QUIT);
-}
-
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.cpp:
```diff
@@ -493,6 +484,15 @@ int CXBMCApp::android_printf(const char *format, ...)
+    StartActivity(getPackageName());
+  }
+}
+
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.cpp:
```diff
@@ -804,13 +804,69 @@ void CXBMCApp::onReceive(CJNIIntent intent)
+  delete event;
+  return ret;
+}
+
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.h:
```diff
@@ -23,6 +23,7 @@
 #include <pthread.h>
 #include <string>
 #include <vector>
+#include <map>
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.h:
```diff
@@ -61,6 +62,25 @@ struct androidPackage
+  int m_resultcode;
+};
+
+
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.h:
```diff
@@ -70,6 +90,7 @@ class CXBMCApp : public IActivityHandler, public CJNIMainActivity,
   virtual ~CXBMCApp();
   virtual void onReceive(CJNIIntent intent);
   virtual void onNewIntent(CJNIIntent intent);
+  virtual void onActivityResult(int requestCode, int resultCode, CJNIIntent resultData);
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.h:
```diff
@@ -101,10 +122,12 @@ class CXBMCApp : public IActivityHandler, public CJNIMainActivity,
   static int android_printf(const char *format, ...);
-  
+  static void BringToFront();
+
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.h:
```diff
@@ -101,10 +122,12 @@ class CXBMCApp : public IActivityHandler, public CJNIMainActivity,
   static bool EnableWakeLock(bool on);
-  static bool HasFocus();
+  static bool HasFocus() { return m_hasFocus; }
+  static bool IsResumed() { return m_isResumed; }
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.h:
```diff
@@ -125,6 +148,8 @@ class CXBMCApp : public IActivityHandler, public CJNIMainActivity,
   static void SetRefreshRate(float rate);
   static int GetDPI();
 
+  static int WaitForActivityResult(const CJNIIntent &intent, int requestCode, CJNIIntent& result);
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.h:
```diff
@@ -167,6 +192,7 @@ class CXBMCApp : public IActivityHandler, public CJNIMainActivity,
   static CJNIWakeLock *m_wakeLock;
   static int m_batteryLevel;
   static bool m_hasFocus;
+  static bool m_isResumed;
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.h:
```diff
@@ -175,14 +201,13 @@ class CXBMCApp : public IActivityHandler, public CJNIMainActivity,
   pthread_t m_thread;
   static CCriticalSection m_applicationsMutex;
   static std::vector<androidPackage> m_applications;
+  static std::vector<CActivityResultEvent*> m_activityResultEvents;
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.h:
```diff
@@ -175,14 +201,13 @@ class CXBMCApp : public IActivityHandler, public CJNIMainActivity,
   static CVideoSyncAndroid* m_syncImpl;
 
-  void XBMC_Pause(bool pause);
-  void XBMC_Stop();
```
no

koying commented on xbmc/platform/android/activity/android_main.cpp:
```diff
@@ -105,6 +105,13 @@ extern "C" JNIEXPORT jint JNI_OnLoad(JavaVM *vm, void *reserved)
+      (void*)&CJNIMainActivity::_onActivityResult
+    };
+    env->RegisterNatives(cMain, &mOnActivityResult, 1);
+
```
no

koying commented on xbmc/platform/android/jni/Activity.cpp:
```diff
@@ -41,3 +41,11 @@ bool CJNIActivity::moveTaskToBack(bool nonRoot)
+    "startActivityForResult", "(Landroid/content/Intent;I)V",
+    intent.get_raw(), requestCode);
+}
+
```
no

koying commented on xbmc/platform/android/jni/Activity.h:
```diff
@@ -32,6 +33,7 @@ class CJNIActivity : public CJNIContext
   ~CJNIActivity();
 
   static bool moveTaskToBack(bool nonRoot);
+  static void startActivityForResult(const CJNIIntent &intent, int requestCode);
```
none of this

koying commented on xbmc/platform/android/jni/ContentResolver.cpp:
```diff
@@ -45,5 +45,17 @@ CJNICursor CJNIContentResolver::query(const CJNIURI &uri, const std::vector<std:
+CJNIList<CJNIURIPermission> CJNIContentResolver::getPersistedUriPermissions()
+{
+  return call_method<jhobject>(m_object,
+    "getPersistedUriPermissions", "()Ljava/util/List;");
```
no

koying commented on xbmc/platform/android/jni/ContentResolver.h:
```diff
@@ -20,16 +20,20 @@
+  CJNIList<CJNIURIPermission> getPersistedUriPermissions();
 
   static void PopulateStaticFields();
   static std::string SCHEME_CONTENT;
```
none

koying commented on xbmc/platform/android/jni/Context.cpp:
```diff
@@ -91,6 +93,8 @@ void CJNIContext::PopulateStaticFields()
+  CJNIDocument::PopulateStaticFields();
 }
 
 CJNIPackageManager CJNIContext::GetPackageManager()
```
none

koying commented on xbmc/platform/android/jni/Document.cpp:
```diff
@@ -0,0 +1,42 @@
+  CJNIDocument::COLUMN_SIZE                  = jcast<std::string>(get_static_field<jhstring>(c,"COLUMN_SIZE"));
+  CJNIDocument::COLUMN_FLAGS                 = jcast<std::string>(get_static_field<jhstring>(c,"COLUMN_FLAGS"));
+  CJNIDocument::MIME_TYPE_DIR                = jcast<std::string>(get_static_field<jhstring>(c,"MIME_TYPE_DIR"));
+}
```
none

koying commented on xbmc/platform/android/jni/Document.h:
```diff
@@ -0,0 +1,41 @@
+protected:
+  CJNIDocument();
+  ~CJNIDocument(){}
+};
```
none

koying commented on xbmc/platform/android/jni/DocumentsContract.cpp:
```diff
@@ -0,0 +1,59 @@
+{
+  return (CJNIURI)call_static_method<jhobject>("android/provider/DocumentsContract",
+    "buildDocumentUriUsingTree", "(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;", treeUri.get_raw(), jcast<jhstring>(parentDocumentId));
+}
```
none

koying commented on xbmc/platform/android/jni/DocumentsContract.h:
```diff
@@ -0,0 +1,40 @@
+protected:
+  CJNIDocumentsContract();
+  ~CJNIDocumentsContract(){}
+};
```
none

koying commented on xbmc/platform/android/jni/Environment.cpp:
```diff
@@ -24,6 +24,14 @@
+{
+  jhclass c = find_class("android/os/Environment");
+  CJNIEnvironment::MEDIA_MOUNTED = jcast<std::string>(get_static_field<jhstring>(c, "MEDIA_MOUNTED"));
+}
```
no

koying commented on xbmc/platform/android/jni/Environment.h:
```diff
@@ -26,11 +26,15 @@ class CJNIFile;
-  ~CJNIEnvironment(){};
-};
+  ~CJNIEnvironment() {};
+};
```
none

koying commented on xbmc/platform/android/jni/Intent.cpp:
```diff
@@ -25,11 +25,13 @@
+  ACTION_OPEN_DOCUMENT_TREE  = jcast<std::string>(get_static_field<jhstring>(clazz,"ACTION_OPEN_DOCUMENT_TREE"));
 }
 
 CJNIIntent::CJNIIntent(const std::string &action) : CJNIBase("android/content/Intent")
```
none

koying commented on xbmc/platform/android/jni/Intent.h:
```diff
@@ -57,4 +57,5 @@ class CJNIIntent : public CJNIBase
   static void PopulateStaticFields();
   static std::string EXTRA_KEY_EVENT;
+  static std::string ACTION_OPEN_DOCUMENT_TREE;
 };
```
no

koying commented on xbmc/platform/android/jni/List.cpp:
```diff
@@ -47,3 +48,4 @@ template class CJNIList<CJNIScanResult>;
 template class CJNIList<CJNIWifiConfiguration>;
 template class CJNIList<CJNIApplicationInfo>;
 template class CJNIList<CJNIViewInputDeviceMotionRange>;
+template class CJNIList<CJNIURIPermission>;
```
none

koying commented on xbmc/platform/android/jni/Makefile.in:
```diff
@@ -60,6 +60,9 @@ SRCS      += DisplayMetrics.cpp
+SRCS      += Document.cpp
 
 LIB        = jni.a
 
```
no

koying commented on xbmc/platform/android/jni/URIPermission.cpp:
```diff
@@ -0,0 +1,36 @@
+{
+  return (CJNIURI)call_method<jhobject>(m_object,
+    "getUri", "()Landroid/net/Uri;");
+}
```
none

koying commented on xbmc/platform/android/jni/URIPermission.h:
```diff
@@ -0,0 +1,37 @@
+
+private:
+  CJNIURIPermission();
+};
```
none

koying commented on xbmc/video/VideoDatabase.cpp:
```diff
@@ -3480,6 +3480,32 @@ void CVideoDatabase::GetDetailsFromDB(const dbiplus::sql_record* const record, i
+
+  return details;
+}
+
```
Can't remember ;)
I think it's needed

razzeee replied:
It is :)

koying commented on xbmc/video/VideoDatabase.h:
```diff
@@ -835,6 +836,7 @@ class CVideoDatabase : public CDatabase
   void ClearMovieSet(int idMovie);
   void SetMovieSet(int idMovie, int idSet);
   bool SetVideoUserRating(int dbId, int rating, const MediaType& mediaType);
+  CVideoInfoTag GetDetailsByTypeAndId(VIDEODB_CONTENT_TYPE type, int id);
```
see above

razzeee replied:
That above was a comment, I just removed. So it should be fine?

koying replied:
Well, if the implem above is needed, so is the declaration :D

koying commented on xbmc/windowing/WinSystem.h:
```diff
@@ -82,6 +82,7 @@ class CWinSystemBase
   virtual bool Restore() { return false; }
   virtual bool Hide() { return false; }
   virtual bool Show(bool raise = true) { return false; }
+  virtual bool BringToFront() { return false; }
```
no

koying commented on xbmc/windowing/egl/EGLNativeType.h:
```diff
@@ -146,6 +146,9 @@ class CEGLNativeType
 
+/*! \brief Bring the current window to front */
+  virtual bool  BringToFront() = 0;
+
```
no

koying commented on xbmc/windowing/egl/EGLNativeTypeAmlogic.h:
```diff
@@ -46,6 +46,7 @@ class CEGLNativeTypeAmlogic : public CEGLNativeType
   virtual bool  GetPreferredResolution(RESOLUTION_INFO *res) const;
 
   virtual bool  ShowWindow(bool show);
+  virtual bool  BringToFront() { return false; }
```
no

koying commented on xbmc/windowing/egl/EGLNativeTypeAndroid.cpp:
```diff
@@ -227,3 +227,8 @@ bool CEGLNativeTypeAndroid::ShowWindow(bool show)
+bool CEGLNativeTypeAndroid::BringToFront()
+{
+  CXBMCApp::BringToFront();
+}
```
no

koying commented on xbmc/windowing/egl/EGLNativeTypeAndroid.h:
```diff
@@ -46,4 +46,6 @@ class CEGLNativeTypeAndroid : public CEGLNativeType
 
   virtual bool  ShowWindow(bool show);
+  virtual bool  BringToFront();
+
```
no

koying commented on xbmc/windowing/egl/EGLNativeTypeIMX.h:
```diff
@@ -55,6 +55,7 @@ class CEGLNativeTypeIMX : public CEGLNativeType
   virtual bool  GetPreferredResolution(RESOLUTION_INFO *res) const;
 
   virtual bool  ShowWindow(bool show);
+  virtual bool  BringToFront() { return false; }
```
no

koying commented on xbmc/windowing/egl/EGLNativeTypeRaspberryPI.h:
```diff
@@ -52,6 +52,8 @@ class CEGLNativeTypeRaspberryPI : public CEGLNativeType
 
   virtual bool  ShowWindow(bool show);
+  virtual bool  BringToFront() { return false; }
+
```
no

koying commented on xbmc/windowing/egl/EGLWrapper.cpp:
```diff
@@ -196,6 +196,14 @@ bool CEGLWrapper::ShowWindow(bool show)
+
+  return m_nativeTypes->BringToFront();
+}
+
```
no

koying commented on xbmc/windowing/egl/EGLWrapper.h:
```diff
@@ -41,6 +41,7 @@ class CEGLWrapper
   bool SetNativeResolution(RESOLUTION_INFO& res);
   bool ProbeResolutions(std::vector<RESOLUTION_INFO> &resolutions);
   bool ShowWindow(bool show);
+  bool BringToFront();
```
no

koying commented on xbmc/windowing/egl/WinSystemEGL.cpp:
```diff
@@ -496,6 +496,11 @@ bool CWinSystemEGL::Show(bool raise)
+{
+  return m_egl->BringToFront();
+}
+
```
no

koying commented on xbmc/windowing/egl/WinSystemEGL.h:
```diff
@@ -56,6 +56,8 @@ class CWinSystemEGL : public CWinSystemBase, public CRenderSystemGLES
   virtual bool  Hide();
   virtual bool  Show(bool raise = true);
+  virtual bool  BringToFront();
+
```
no

koying commented on xbmc/windows/GUIWindowFileManager.cpp:
```diff
@@ -482,6 +482,13 @@ bool CGUIWindowFileManager::Update(int iList, const std::string &strDirectory)
+    pContentItem->SetArt("thumb", "DefaultFolder.png");
+    pContentItem->SetLabelPreformated(true);
+    m_vecItems[iList]->Add(pContentItem);
+#endif
```
no

koying commented on xbmc/platform/android/activity/XBMCApp.cpp:
```diff
@@ -804,10 +794,36 @@ void CXBMCApp::onReceive(CJNIIntent intent)
+      params.push_back(targeturl.Get());
+      params.push_back("return");
+      CApplicationMessenger::GetInstance().PostMsg(TMSG_GUI_ACTIVATE_WINDOW, WINDOW_MUSIC_NAV, 0, nullptr, "", params);
+    }
```
This "semi-hack" allows to show the info page, rather than launching playback

razzeee replied:
This part I understand, the "android.intent.action.VIEW" part is what I'm curious about. As I have commented out some stuff there ;)

koying replied:
Well, the VIEW intent is for playing the file immediately.
It's been there for a while. Allows you to use Kodi as a target player when selecting a video in an android file browser

razzeee replied:
Then I'll drop it as it's unrelated to leanback

koying commented on xbmc/utils/URIUtils.cpp:
```diff
@@ -211,6 +211,17 @@ void URIUtils::Split(const std::string& strFileNameAndPath,
+    else i--;
+  }
+  if (i > 0)
+    strFileName = strFileName.substr(0, i);
```
That's a fix for the fact that split was not taking options ("?") into account, leading to filenames like "blabla.ext?key=value"

razzeee replied:
yeah, I think I was confused by strFileName
`std::string& strFileName` is no pointer but a reference so it does get changed, correct?

garbear replied:
also, if you're chopping off the end of a string, use `string::erase()`

razzeee committed:
Add more split tests

Razzeee commented:
This is now trimmed down and I started to do some improvements, what's not working yet:

- [ ] Recommendations for movies open the movie info, but it's empty
- [ ] If I'm in the movie info and try another recommendation for e.g. tvshows it will fail
- [ ] Recommenations get refreshed and reordered quiet often

For later:

- [ ] Start kodi on system start but only on tvs

Razzeee commented:
PS: The test class is untested, as test suit is broken atm

Razzeee commented:
As the movie info is a modal dialog, trying to send a new window gets refused.

```
Kodi    : Debug Print: Activate of window '10502' refused because there are active modal dialogs
```

razzeee committed:
Trying to fix showinfo

razzeee commented on tools/android/packaging/xbmc/src/org/xbmc/kodi/XBMCJsonRPC.java.in:
```diff
@@ -0,0 +1,647 @@
+
+  private static String TAG = "@APP_NAME@json";
+
+  private String m_jsonURL = "http://localhost:8080";
```
@SyncedSynapse is there an easy way to get the port kodi is listening on?

SyncedSynapse replied:
Not that i know of. In the remote the user is supposed to validate it.

Edit: using Zeroconf you can get the port, but i don't think you can assume it is available and enabled.

da-anda replied:
what about using the socket connection to JSON instead of http? Would that help?

koying replied:
@da-anda You also need to "download" thumb+fanart.
Can that be done via a socket connection? Didn't check but I doubt it.

Regarding port configuration, the "easy" way (for us) is to do it via xbmc_env.properties.
Maybe @SyncedSynapse and/or @poisdeux could create a companion app to configure xbmc_env (or integrate it in Kore?)

da-anda replied:
what about moving the leanback integration in general into a companion app? That app could then also autostart Kodi on startup (if that's possible). Benefit would also be that improvements can be done outside of release cycles

poisdeux replied:
If I understand correctly, you want to be able to start Kodi remotely on an Android device using Kore or another app? If so, Network Service Discovery might be a solution:
http://developer.android.com/training/connect-devices-wirelessly/nsd.html

razzeee replied:
No, the problem is knowing the correct port without asking the user.

koying replied:
Don't forget u/p, if applicable

da-anda replied:
what about using TCP connection (port 9090) for the basic communication with Kodi as well as reading it's HTTP port setting (https://github.com/xbmc/xbmc/blob/master/xbmc/interfaces/json-rpc/SettingsOperations.cpp#L204) and only grab thumbs via HTTP GET?

koying replied:
Users can also change the TCP port , although less easily: http://kodi.wiki/view/Advancedsettings.xml#.3Cjsonrpc.3E

da-anda replied:
well, then move everythinig out to a companion app "Kodi Leanback Experience" (or alike) and make it configurable in case anybody messed with as.xml etc. I don't think we can and should add the needed logic to Kodi itself (except ofc for the probably required interfaces)

razzeee replied:
Let me quickly draft a payed app for that :smile:

koying replied:
Why moving to a separate app? You can have more than 1 launchable activity inside a single app.
You could create a "settings" activity inside the Kodi app if you want more than xbmc_env manual editing, assuming the json settings would be added to it.

da-anda replied:
Separate app, because a) you can improve it independet of release cycles (which might be needed), b) can make it autolaunch Kodi on startup, c) find a pure Java dev as maintainer, d) more easily share code with Kore, ...

At least in my opinion this could work very well as standalone companion app. If you guys think it's better suited as part of Kodi itself, I'm not against doing it that way, but given that we remote access the main Kodi process anyways and that it might be a bit tricky to add a Android based config UI for ports etc (which appears to be needed), it just feels wrong as part of the main app. If we won't need any user config options, then I'd of course prefer it being part of Kodi itself.

koying replied:
a) is a poor excuse to do a separate app, imho ;)  (I personally find 3 months to do a point release hardly RERO)
b) well, yeah, as much as Kodi can be autostarted
c) is irrelevant (you can just plug the activity developed by "a pure java developper" into Kodi in no time 
d) see c)

The biggest gripe I have with the separate app approach is that one is already existing, really...
But yeah, why not, if @SyncedSynapse or @poisdeux are up for it...

razzeee replied:
Well b is also solveable via one app

da-anda replied:
maybe let's focus on the detection of the webservice port first and go from there. If we can do it without user interaction/config, having it in one app is the way to go anyways. If we really need config options, we can decide on the way to go.

And I'd not put this feature into Kore - I at least don't see it fit into Kore, as Kore is an app to control a remote Kodi installation while this feature is only relevant for AndroidTV and Kodi on the same system. But as said, that's just my opinion and you're free to do what you guys think is best.

koying replied:
> If we can do it without user interaction/config

~~Well, the whole point is that it's not (per definition regarding user/password).~~ Well, we _could_ read the guisettings directly, but that's kind of ugly...
I think step one is to add those config to xbmc_env. I'll add it to SPMC, so you can backport would you chose to follow the same path.

da-anda replied:
and reading the settings via TCP is not an option? That way we only would have to expose the TCP port (which most won't change anyways in as.xml) and could use JSON-RPC to grab all other settings (http port, user, pass)

laurimyllari committed:
colormanager: strip extra debug fixup

laurimyllari committed:
colormanager: add override keyword to virtual methods

laurimyllari committed:
settings: clean up colormanager settings

MartijnKaijser committed:
[droid] MediaCodec API21+ buffer handling. thx to koying

Rechi committed:
[depends] bump mDNSResponder to 625.41.2

da-anda commented on tools/android/packaging/xbmc/src/org/xbmc/kodi/XBMCJsonRPC.java.in:
```diff
@@ -0,0 +1,647 @@
+        JSONObject req = request_object(String.format(SEARCH_MOVIES_JSON, /*"\"operator\": \"contains\", \"field\": \"title\", \"value\": \"" + query + "\"", limit));*/
+        "\"or\": [" +
+        "{\"operator\": \"contains\", \"field\": \"title\", \"value\": \"" + query + "\"}," +
+        "{\"operator\": \"contains\", \"field\": \"set\", \"value\": \"" + query + "\"}," +
```
I think we should also search in the tagline and likely also original title

razzeee replied:
I can add that

koying replied:
Be careful, though.
The more fields you add, the longer it will take for search to return results (already too long for my taste ;)).
originaltitle is a natural, but tagline = increased irrelevant result and a poor field for searching, imo.

If you know neither the title, nor the original title nor one of the actors of a movie, I doubt tagline will help ;)

razzeee replied:
Yep, sounds better, was a bit sceptic about tagline

da-anda replied:
the reason why I suggested tagline was that regardless how hard I tried pronouncing a specific movie title, I never got a match because Google does a poor job with detecting English words when voice search is set f.e. to German. And IIRC that movie had an English name also in German, so the only way to get a match was the tagline, which worked fine with Google Play Movies. Once I switched voice search language to English, movie was detected just fine, but then none of the German titled movies where found.

koying replied:
If both (localized) title and originaltitle are searched, that should solve your issue, shouldn't it?
On a side note, I have voice search set to french and I'm quite happy with the recognition rate of english words...

da-anda replied:
not if localized title is also English - and that was the issue with this particular movie. Unfortunately I don't remember which movie exactly I had issues with. And it also felt strange to pronounce "Terminator" with a german accent in order for Google to detect it correctly. But if it's really that much of a performance hit, drop it

Tolriq replied:
@Razzeee  Why do you need JSON at all to communicate with Kodi from inside Kodi ?

Asking a user to activate web server to open it's Kodi to outside world (Knowing how unsecure webserver is by nature) to have TV recommendation on the same device that Kodi is installed sounds not cool. (Specially without a default password set)

TCP 9090 on localhost is on by default and getting the port is only eventually adding a map in JNI. Same for thumb directory. Since same process no security problem at all.

Furthermore adding 2 or 3 required JNI functions to manage this would be way more efficient than JSON encoding / decoding that is quite slow on low end devices.

razzeee replied:
I don't think I can answer that, as I only cherrypicked from @koying and tried to improve some parts. Pretty new to doing android stuff, so if you can point me in the right directions to improve it even more, that would be appreciated (not sure how much time I will have in the next months)

Tolriq replied:
Will try to look a little more how Kodi for Android is built and see what I can do more.

razzeee replied:
Thank you

koying committed:
CHG: [leanback] force recommendation update every hour

Tolriq commented on tools/android/packaging/xbmc/src/org/xbmc/kodi/Main.java.in:
```diff
@@ -182,6 +183,47 @@ public class Main extends NativeActivity implements Choreographer.FrameCallback
+              e.printStackTrace();
+            }
+          }
+        }.start();
```
This is awful :)

On each activity resume you start a new thread that update leanback every 10 seconds.

Meaning you start kodi then move to other app 2 or 3 times and now have 4 threads doing the same thing.

Thread should be started in onCreate if needed with an handler, then you send messages to the handler to trigger a refresh when needed. (Not sure every 10 second is a good value).

koying replied:
> On each activity resume you start a new thread that update leanback every 10 seconds

Nah. Those are one-shot threads (no loop) and mandatory because Android refuses to do httprequests on the main thread (even runnables don't work, as they run with original thread affinity with handler.postdelayed). The 10 sec delay is for leaving Kodi time to settle on first run.

Tolriq replied:
Ho ok :( 

But then in current code there's no update of recommendations after something is watched or library scraped ?

koying replied:
Indeed. Started working on that https://github.com/koying/SPMC/commits/leanbackUpdate.
This just do an hourly update, but updating on playback end/stop/update is a good idea, although tricky on a mysql/multiple devices environment...

Tolriq replied:
Well if you keep JSON for internal communication (Why JSON and not native as you have the necessary skills for that ?), using TCP and listening for notifications would handle 95% of the basic cases. Scheduled updates can fill the rest.

koying replied:
> Why JSON and not native as you have the necessary skills for that ?

1) That's what JSON is for
2) Overkill for a "nice-to-have" like recommendations, imo
3) That's additional  code nobody but an Android/JNI specialist will be able to maintain
4) I'm lazy ;)

Tolriq replied:
Well  #1 would be true if there was a way to have HTTP server only on localhost :( Users are not really aware of the risk of the webserver with bad configuration like possible exposing of their passwords.

Securing WebServer / JSON was long discussed but never approached :(

koying replied:
> Well #1 would be true if there was a way to have HTTP server only on localhost

True, true. I'm looking at having the webserver on localhost ON by default to avoid the "i-don't-have-recommendations" tickets ;)
@mrmc already closed the password gap, iirc.

FernetMenta committed:
VideoPlayer: expose stream player info to GUI

laurimyllari committed:
colormanager: don't use singleton

laurimyllari committed:
settings: kodify copyright header

koying committed:
FIX: [leanback] notification id's + tweaks

Conflicts:
	tools/android/packaging/xbmc/src/org/xbmc/kodi/Main.java.in
	xbmc/android/activity/XBMCApp.cpp

fetzerch committed:
[cmake] Increase link performance (GOLD, DebugFission)

Increase the link performance with GCC by
- defaulting to GNU Gold Linker if available
- defaulting to Debug Fission if available (https://gcc.gnu.org/wiki/DebugFission)

Some measurements on Ubuntu Xenial (GNU GCC 5.3.1, Binutils 2.26)
with incremental linking (touch xbmc/FileItem.cpp) of a debug build:
- GNU ld:   16.5s
- GNU gold:  6.5s

popcornmix committed:
omxvideo: Remove call to AutoInterlaceMethod. Treat auto as advanced

laurimyllari committed:
colormanager: use m_ prefix for members

laurimyllari committed:
colormanager: fix build without lcms2

laurimyllari committed:
gl: remove unnecessary null check

laurimyllari committed:
gl: use unique_ptr for ColorManager

laurimyllari committed:
gl: don't use vertical alignment

laurimyllari committed:
dialog: don't check obsolete HAS_VIDEO_PLAYBACK

laurimyllari committed:
settings: move cms settings to the right group

laurimyllari committed:
gl: Try to unbreak GLES builds

laurimyllari committed:
dialog: Try to unbreak Windows builds

FernetMenta committed:
update XCode for 3dlut

popcornmix committed:
mmal_codec: Use EOS through codec to determine drain is complete

Rather than relying on a timeout from codec, feed an EOS through to ensure all frames have been returned

popcornmix committed:
rbp: Update transposed video scaling to match other platforms

popcornmix committed:
mmalcodec: Add another buffer when deinterlacing

See: http://forum.kodi.tv/showthread.php?tid=276372

ksooo committed:
[PVR] Timers: Allow to change timer type for existing timers.

laurimyllari committed:
dialog: do not build CMS settings dialog unless HAS_GL

peak3d committed:
Fix double mem alloation if streamcodec hasn't changed

FernetMenta committed:
Merge pull request #10018 from mapfau/memleak

VIdeoPlayer: Fix double mem alloation if streamcodec hasn't changed

phate89 committed:
Make default rating private allowing to set it only with a set rating

phate89 committed:
[breaking change] Change JSON ratings structure to match art one

ksooo committed:
[PVR] Fix epg tag <-> timer tag association handling.

ronie committed:
[Estouchy] update

ronie committed:
Merge pull request #10021 from ronie/estouchy17

[Estouchy] update

FernetMenta committed:
remove DataCacheCore from systemGlobals

fritsch committed:
AESinkPULSE: Support 384 khz with PA Server 9.0

ksooo committed:
[PVR] CPVRTimerInfoTag: refactor ctors.

laurimyllari committed:
settings: color->colour for proper English

laurimyllari committed:
dialog: hide CMS OSD if not available

notspiff committed:
added: FFmpeg-supplied RTMP support

notspiff committed:
quell redefine warning in cmake build

notspiff committed:
quell deprecation warning in microhttpd code

notspiff committed:
call avformat_network_init()

quells a runtime error and adds forwards compatibility (it will become
mandatory eventually)

also should be slightly more efficient as global state is initialized
exactly once instead of once per connection

Paxxi committed:
Merge pull request #10023 from notspiff/quell_warnings_2

Quell some warnings

phate89 committed:
Merge pull request #10020 from phate89/ratings_as_art

Change ratings to reflect json art (and future uniqueid) structure

Montellese committed:
[settings] remove non-existing settings from code

Montellese committed:
[upnp] CUPnP::InvokeUpdateObject: fix case when MediaServer does not support UpdateObject() action

ksooo committed:
[PVR] EPG Grid: fix wrong status icons for epg events.

ksooo committed:
[PVR] Guide window: refactor obtaining grid control.

popcornmix committed:
webserver: fixup breakage in #10023

popcornmix committed:
Merge pull request #10029 from popcornmix/quellfix

webserver: fixup breakage in #10023

popcornmix committed:
rbp: Update to use new processInfo data cache

ksooo committed:
Merge pull request #10024 from ksooo/pvr-fix-epgtag-timer-assoc

[PVR] Fix epg tag <-> timer tag association handling.

jenkins4kodi committed:
Merge pull request #10022 from fritsch/pulse384

vzahradnik committed:
Adding keyboard layout: Slovak QWERTZ, QWERTY, ABC

bkuhls committed:
configure.ac: Fix typo in help text of non-free option

bkuhls committed:
ZeroconfMDNS.cpp: add header to fix compile with gcc 4.9 & uClibc

Without the patch compilation fails with
ZeroconfMDNS.cpp:133:139: error: ‘htons’ was not declared in this scope

Infos about the toolchain used:
gcc version 4.9.3 (Buildroot 2016.05-00002-g5dabb45), uClibc-NG 1.0.15

jenkins4kodi committed:
Merge pull request #10027 from Montellese/upnp_fix_resume

MartijnKaijser committed:
Merge pull request #10034 from bkuhls/typo

configure.ac: Fix typo in help text of non-free option

ksooo committed:
[PVR] Guide window: fix timer icons not displayed after scheduling a recording from inside guide info dialog opened from inside guide window.

ksooo committed:
[PVR] PVR windows: Fix observer handling.

MartijnKaijser committed:
Merge pull request #10035 from bkuhls/mdns

ZeroconfMDNS.cpp: add header to fix compile with gcc 4.9 & uClibc

ksooo committed:
[PVR] Cleanup: Change CGUIWindowPVRBase::SetGroup param from value to const reference.

ksooo committed:
Merge pull request #10036 from ksooo/pvr-fix-timer-icon-issues

[PVR] Guide window: Fix timer icon issues. Fixes #16621

fritsch committed:
FFmpeg: Bump to version 3.1.1-Krypton-Alpha3

ksooo committed:
[PVR] Timers: Assign timers to all matching epg tags, not only to the first. A timer may span over multiple epg events.

FernetMenta committed:
Revert "[addons] Preserve strings which are not string ids in enum lvalues"

This reverts commit c2bd593649ed2493e3502ba15e62c62b1bdbe73f.

This breaks a lot of addon using numerical values in enums. This change
should be resubmitted when the issue is fixed

MartijnKaijser committed:
Merge pull request #10040 from FernetMenta/8271

Revert "[addons] Preserve strings which are not string ids in enum lv…

wsnipex committed:
[depends] bump python to 2.7.12

ksooo committed:
[EPG] Cleanup: CEpgContainer: Remove senseless virtual from several member functions. Change some params from value to reference.

popcornmix committed:
mmaldec: Enable zero copy on decoder input buffers

ksooo committed:
[PVR] Cleanup: Rename repeating timer to timer rule: step two: source code (step
one was already done, was renaming at UI level)

ksooo committed:
Merge pull request #10042 from ksooo/pvr-multi-epgtag-timers

[PVR] Guide window: Fix timer icons for manual (not epg-tag based) timers

afedchin committed:
[VideoSyncD3D] CVideoSyncD3D should implement IDispResource interface instead of ID3DResource.

MartijnKaijser committed:
updated splash for Krypton alphas

MartijnKaijser committed:
bump to 17.0 alpha3

MartijnKaijser committed:
Merge pull request #10047 from MartijnKaijser/alpha3

bump to 17.0 alpha3

MartijnKaijser committed:
Merge pull request #10033 from vladimir-zahradnik/feature-sk-keymap

Keyboard layout: Slovak QWERTZ, QWERTY, ABC

MartijnKaijser committed:
Merge pull request #10025 from Rechi/depends/mDNSResponder

[depends] bump mDNSResponder to 625.41.2

MartijnKaijser committed:
Merge pull request #9911 from MartijnKaijser/buffer

[droid] MediaCodec API21+ buffer handling

MartijnKaijser committed:
Merge pull request #10041 from wsnipex/bump-python

[depends] bump python to 2.7.12

jenkins4kodi committed:
Merge pull request #10048 from MartijnKaijser/libxslt

FernetMenta committed:
Merge pull request #9976 from FernetMenta/codecinfo

VideoPlayer: expose stream player info to GUI

FernetMenta committed:
Merge pull request #10015 from notspiff/rtmp_internal

added: FFmpeg-supplied RTMP support

FernetMenta committed:
Merge pull request #9731 from laurimyllari/v17-3dlut

3dlut and simple ICC linking support for color correction

ksooo committed:
Merge pull request #10046 from ksooo/pvr-rename-repeating-timer-to-timer-rule

[PVR] Cleanup: Make timer internal terminology consistent, step 2

hudokkow committed:
[eventclients] Nuke J2ME

jenkins4kodi committed:
Merge pull request #10026 from Montellese/settings_cleaning

MartijnKaijser committed:
[depends] libbluray 0.9.3

MartijnKaijser committed:
[depends] libomxil-bellagio is not used

jenkins4kodi committed:
Merge pull request #10055 from MartijnKaijser/libomxil

MartijnKaijser committed:
Merge pull request #9749 from Tolriq/json_picture_screensaver

Fix screensaver issue with JSON and pictures

tamland committed:
[addons] refactor addon.xml loading from zip files

use the same procedure for local and repository zips

tamland committed:
[addons] remove mutable workarounds

tamland committed:
[addons] drop support for non-zipped and multi-directory repositories

tamland committed:
[addons] properly construct addon instances from repository xml using builders

tamland committed:
[addons] log checksums on mismatch

popcornmix committed:
Merge pull request #10050 from popcornmix/transposescale

rbp: Update transposed video scaling to match other platforms

popcornmix committed:
Merge pull request #10052 from popcornmix/deintbuf

mmalcodec: Add another buffer when deinterlacing

ksooo committed:
[PVR] Observers/Observables: Fix pvr windows <-> pvr components lifecycle problem.

tamland committed:
[addons] fix default sort order in recently updated

Commit:
[settings] Adjustments to the location of settings and defaults

Commit:
[strings] update labels for settings adjustments

MartijnKaijser committed:
Merge pull request #10011 from jjd-uk/settings_defaults

[settings] Adjustments to the location of settings and defaults

MartijnKaijser committed:
Merge pull request #10062 from tamland/fix_addons_sortorder

[addons] fix default sort order in recently updated

ksooo committed:
Merge pull request #10061 from ksooo/pvr-messaging

[PVR] Observers/Observables: Fix pvr windows <-> pvr components lifecycle problem.

Commit:
Added controls for Next item/Prev item in video playback with EDL
(commercials) to go to the end of the next commercial block, or
prev to the beginning of the previous commercial block.

popcornmix committed:
Merge pull request #10053 from popcornmix/deintadv

omxvideo: Remove call to AutoInterlaceMethod. Treat auto as advanced

popcornmix committed:
Merge pull request #10051 from popcornmix/eosdrain

mmal_codec: Use EOS through codec to determine drain is complete

vzahradnik committed:
[PeripheralCecAdapter] Fix: Slovak language code (slo) considered as Slovenian (slv)

Memphiz committed:
VideoPlayer: - fixed c&p - return the correct video processing info when requested

FernetMenta committed:
Merge pull request #10065 from Memphiz/fix_video_process_guiinfo

VideoPlayer: - fixed c&p - return the correct video processing info w…

FernetMenta committed:
VideoPlayer: fix setting pixel format for ffmpeg

FernetMenta committed:
change gui info from Player_Process.xxx to Player.Process(%s)

fetzerch committed:
[cmake] Add CMakeHelpers (utilities for debugging CMake scripts)

fetzerch committed:
[cmake] Opt out if platform configuration cannot be found

For example on Darwin platforms this can indicate a missing toolchain
file.

FernetMenta committed:
Merge pull request #10039 from fritsch/ffmpeg_3.1

FFmpeg: Bump to version 3.1

FernetMenta committed:
VideoPlayer: write audio related process info to cache

FernetMenta committed:
player processInfo: expose integer values as strings

FernetMenta committed:
VideoPlayer: update videodecoder name in processInfo for ffmpeg based codecs

txtranslation committed:
[lang][kodi.core] automatic syntax corrections for the en_GB language file

txtranslation committed:
[lang] updated language files from Transifex

txtranslation committed:
[lang][skin.estuary] updated language files from Transifex

txtranslation committed:
[lang][skin.estouchy] updated language files from Transifex

hudokkow committed:
Merge pull request #10049 from hudokkow/eventclient_cleanup

[eventclients] Nuke J2ME

jenkins4kodi committed:
Merge pull request #10056 from MartijnKaijser/libbluray

FernetMenta committed:
VideoPlayer: set ffmpeg yadif filters in processInfo

tamland committed:
Merge pull request #10068 from LS80/addon_settings_lvalues

[addons] Preserve strings which are not string ids in enum lvalues

fetzerch committed:
[cmake] Print CMake version and add some warnings

CMake 3.5.1 suffers from a crash at configure time that happens on some
Systems: http://cmake.org/Bug/view.php?id=16044
This is fixed in 3.5.2.

Darwin requires CMake at least 3.4 or the usage of the patched version
in depends.

wiromare committed:
CRenderManager: fix GetAspectRatio at playback start

ksooo committed:
Merge pull request #10057 from ksooo/pvr-edit-existing-timers

[PVR] Timers: Allow to change timer type for existing timers.

afedchin committed:
[RenderSystemDX] Hook into user mode driver for selected adapter to fix refresh rate issue in FSE mode.

FernetMenta committed:
Merge pull request #10069 from FernetMenta/codecinfo

Player.Process info label changes

fetzerch committed:
[ColorManager] Replace non-standard VLA with vector

Variable-length arrays are not part of C++ and only allowed in C99. While
they are supported as a GCC extension (if -pedantic is not set), they
cannot be used with MSVC.

This commit replaces the VLA in ColorManager by a std::vector.

jenkins4kodi committed:
Merge pull request #10066 from vladimir-zahradnik/bugfix-cec-slovak

FernetMenta committed:
Merge pull request #10071 from wiromare/fix-aspectratio

CRenderManager: fix GetAspectRatio at playback start

FernetMenta committed:
Merge pull request #10064 from thardie/master

Added navigation controls for EDL (commercials) in PVR playback

Commit:
Fix theme xbt loading that broke during cleanup.
Found a race condition that seems to have been hidden and a screwup during cleanup clobbering state.

FernetMenta committed:
vdpau: refcount decoder with surfaces given to ffmpeg, align with vaapi, fixes pixelation after seek

popcornmix committed:
VideoPlayer: Remove usused variables

FernetMenta committed:
VideoPlayer: check for duplicate filenames when opening multi source demuxer

FernetMenta committed:
player processinfo: set higher precision for FPS and DAR

fetzerch committed:
[cmake] Fix export-files when building in-source

In-source builds were broken due to a mistake in the export-files
logic. Even though this commit resolves the problem, it is recommended
building out-of-source.

fetzerch committed:
Merge pull request #10017 from fetzerch/cmake-fix-rtmp

[android/packaging] Sync rtmp removal

FernetMenta committed:
Merge pull request #10076 from FernetMenta/multi

VideoPlayer: check for duplicate filenames when opening multi source …

fetzerch committed:
Revert "[android/packaging] Sync rtmp removal"

FernetMenta committed:
guilib: do not kill plugin script if we are waiting for user input

MaxKellermann committed:
MusicDatabase: use range-based "for"

MaxKellermann committed:
MusicDatabase: use "auto"

MaxKellermann committed:
VideoDatabase: use range-based "for"

MaxKellermann committed:
VideoDatabase: use "auto"

MaxKellermann committed:
DbUrl: use "override"

MaxKellermann committed:
DbUrl: make the base class "private"

Nobody ever casts to the base class.  By making it "private", we
narrow the interface, reducing it to a few explicitly mentioned APIs.
This will allow further optimizations, such as making more methods
non-virtual.

MaxKellermann committed:
utils/UrlOptions: remove explicit std::string initialization to empty

This is implied, and it's cheaper.

MaxKellermann committed:
URL: inline trivial methods

These are so simple that a function call is bigger than its inline.

Not inlining the destructor because there may be a bloat of implicitly
generated code in there.

MaxKellermann committed:
utils/UrlOptions: make several methods non-virtual

There are no overrides.

MaxKellermann committed:
utils/UrlOptions: use range-based "for"

MaxKellermann committed:
utils/UrlOptions: include cleanup

MaxKellermann committed:
utils/UrlOptions: use std::string::clear()

This is cheaper than assigning an empty string.

MaxKellermann committed:
utils/Variant: include cleanup

MaxKellermann committed:
utils/Variant: use std::vector::emplace_back()

MaxKellermann committed:
DbUrl: include cleanup

MaxKellermann committed:
utils/UrlOptions: use "auto"

MaxKellermann committed:
test/TestBasicEnvironment: include cores/DataCacheCore.h

Needed because gcc 4.8 complains about the use of std::unique_ptr<>
with a forward-declared type:

 /home/travis/build/xbmc/xbmc/xbmc/ServiceManager.h:47:7:   required from here
 /usr/include/c++/4.8/bits/unique_ptr.h:65:22: error: invalid application of ‘sizeof’ to incomplete type ‘CDataCacheCore’

Newer gcc versions do not have this problem, but unfortunately, the
Kodi build server runs this old version.

afedchin committed:
[dependencies] Added EasyHook to package list.

afedchin committed:
Merge pull request #10082 from afedchin/d3d_vsync_fix

[VideoSyncD3D] CVideoSyncD3D should implement IDispResource interface…

Fneufneu committed:
[cmake/freebsd] build Sinks/AESinkOSS

Fneufneu committed:
[cmake/freebsd] remove ldt_keeper from build

like it was done in Makefile before

Fneufneu committed:
[cmake] fix ffmpeg freebsd build

reuse linux args

Fneufneu committed:
[cmake] do not hardcode make command

use CMAKE_MAKE_PROGRAM variable to fix FreeBSD build

Fneufneu committed:
[cmake] also try to find swig3.0

On FreeBSD the binary are: swig2.0 or swig3.0
so try to find version 3.0 if the 2.0 is not there

stefansaraev committed:
[depends] build curl without librtmp support

kodi now uses ffmpeg's own rtmp implementation, and optionaly inputstream.rtmp
rtmp support was picked automaticlay by libcurl configure if librtmp header/libs
are present in sysroot. no point linking our libcurl against librtmp now

jenkins4kodi committed:
Merge pull request #10078 from fetzerch/cmake_fixes_warnings

FernetMenta committed:
Merge pull request #10075 from FernetMenta/vdpau

vdpau: refcount decoder with surfaces given to ffmpeg, align with vaa…

FernetMenta committed:
Merge pull request #10080 from FernetMenta/plugin

guilib: do not kill plugin script if we are waiting for user input

fetzerch committed:
Merge pull request #10079 from xbmc/revert-10017-cmake-fix-rtmp

Revert "[android/packaging] Sync rtmp removal"

fetzerch committed:
Merge pull request #9924 from fetzerch/cmake_gold

[cmake/linux] Increase link performance (GOLD, DebugFission)

fetzerch committed:
[cmake/cpluff] Add generated artefacts to .gitignore

With CMake, cpluff is still built in the source directory.
- Ignore dwarf objects files (130d154afdc13101f7265356086beedbdaeb8e7f)
- Ignore cpluff-console and cpluff-loader

Fneufneu committed:
[cmake/freebsd] Update freebsd treedata

It's the linux file + xbmc/freebsd

Fneufneu committed:
[cmake/freebsd] Use linux Install.cmake

FernetMenta committed:
Merge pull request #10077 from FernetMenta/codecinfo

player processinfo: set higher precision for FPS and DAR

popcornmix committed:
Merge pull request #10084 from popcornmix/vpunused

VideoPlayer: Remove usused variables

jenkins4kodi committed:
Merge pull request #10073 from Paxxi/fix_theme

jenkins4kodi committed:
Merge pull request #10085 from popcornmix/inzerocopy

tamland committed:
PluginDirectory: dont time out while script is modal/not cancelled

fixes regression from 95ffdcc381288c

tamland committed:
Merge pull request #10059 from tamland/repository_refactor

[addons] repository code cleanup

fetzerch committed:
Merge pull request #10089 from fetzerch/cmake_cpluff

[cmake/cpluff] Add generated artefacts to .gitignore

FernetMenta committed:
fix test suite by declaring a deleter for CDataCacheCore

FernetMenta committed:
Revert "vdpau: refcount decoder with surfaces given to ffmpeg, align with vaa…"

FernetMenta committed:
Merge pull request #10092 from xbmc/revert-10075-vdpau

Revert "vdpau: refcount decoder with surfaces given to ffmpeg, align with vaa…"

fetzerch committed:
[cmake/android] Install librtmp

After the removal of RTMP in https://github.com/xbmc/xbmc/pull/10017,
the library fails to get packaged for android because it is no more
populated in the dyload-libs variables.

As the library is still needed (https://github.com/xbmc/xbmc/pull/10079)
we manually add it to the files that we're going to package.

Fixes:
cp: cannot stat 'kodi-build-android/install/lib/librtmp.so': No such file or directory
Makefile:121: recipe for target 'libs' failed

fetzerch committed:
Merge pull request #10094 from fetzerch/cmake_fix_android_packaging

[cmake/android] Install librtmp

FernetMenta committed:
Merge pull request #10090 from FernetMenta/testsuite

fix test suite by declaring a deleter for CDataCacheCore

fetzerch committed:
Merge pull request #10087 from Fneufneu/freebsd_cmake

Freebsd cmake

wsnipex committed:
Merge pull request #10086 from stefansaraev/curl-nortmp

[depends] build curl without librtmp support

FernetMenta committed:
Merge pull request #10097 from FernetMenta/streams

VideoPlayer: update selectionstreams when stream parameters changes

drieschel committed:
Duplicate include removed

drieschel committed:
Removed duplicate switched to right one

Commit:
Remove cached archives at boot

Paxxi committed:
Merge pull request #10099 from drieschel/amlutils_duplicate_include

Duplicate include removed

tmelin committed:
Subtitles/SAMI: fix parsing of files with leading whitespace

SAMI (.smi) subtitle files with indentation in lines were not read correctly.
In case subtitle file has indentation for <> tags, strnicmp() failed.

Trim lines to fix comparing of strings.

Function refactored to use helpers from StringUtils.

Signed-off-by: Tomas Melin <tomas.melin@iki.fi>

ksooo committed:
[PVR] Recordings window: make 'group items' setting persistent.

ksooo committed:
Merge pull request #10101 from ksooo/pvr-group-recordings-setting

[PVR] Recordings window: make 'group items' setting persistent.

garbear committed:
[cmake] Sync Peripheral API

Credit fetzerch, https://github.com/garbear/xbmc/pull/53

FernetMenta committed:
IPlayer: properly sync playspeed with application

Tolriq committed:
Add ipv6 support to Event Server
More than inspired by http://trac.kodi.tv/ticket/12491 with some fixes

FernetMenta committed:
player: drop unneded method ControlsVolume

FernetMenta committed:
applicationPlayer: reset caching timers on OpenFile

tamland committed:
Merge pull request #10093 from tamland/fix_plugin_dir

PluginDirectory: dont time out while script is modal/not cancelled

Paxxi committed:
Merge pull request #9225 from Paxxi/archive

Fix out of memory crash in CArchive

afedchin committed:
[OverlayRendererDX] Fix rendering overlays in TAB/SBS modes.

FernetMenta committed:
Merge pull request #10103 from FernetMenta/state

IPlayer: properly sync playspeed with application

FernetMenta committed:
VideoPlayer: DVDMessageQueue, round up instead to the nearest integer

FernetMenta committed:
VideoPlayer: set clock to video pts when seeking while player is paused

afedchin committed:
Merge pull request #10105 from afedchin/dx_overlay_fix_3d

[OverlayRendererDX] Fix rendering overlays in TAB/SBS modes.

Commit:
Update usage of Crc32 after changing methods to static

afedchin committed:
Merge pull request #10083 from afedchin/d3d11_refresh_hack

[D3D11] Hack to fix refresh rate issue in full screen exclusive mode.

Paxxi committed:
Merge pull request #10107 from Paxxi/fix_crc32

Update usage of Crc32 after changing methods to static

Jalle19 committed:
Merge pull request #10100 from tmelin/smi-parse-fix

Subtitles/SAMI: fix parsing of files with leading whitespace

FernetMenta committed:
Merge pull request #10104 from Tolriq/event_server_ipv6

Add ipv6 support to Event Server

koying committed:
FIX: [droid] high cpu with amc surface

FernetMenta committed:
Merge pull request #10109 from FernetMenta/seek

VideoPlayer: fix seeking while paused

MartijnKaijser committed:
Merge pull request #10070 from MaxKellermann/opt1

Various micro-optimizations in the URL code

MartijnKaijser committed:
fix typo after 093912e311525de7bf39f316b8cfa20e400febad

jenkins4kodi committed:
Merge pull request #10111 from fritsch/androidfixes

FernetMenta committed:
AE: fix resetting rr to 1.0

Commit:
Missed this one

Paxxi committed:
Merge pull request #10113 from Paxxi/fix_test

[fix][tests] Missed this one

MartijnKaijser committed:
[win32] update AppVeyor with ffmpeg 3.1.1

MartijnKaijser committed:
[win32] Improve installer and add appxmanifest

MartijnKaijser committed:
[fix] revert commit c9884248b729709450087f05dd8449e794253a6 utils/Variant: use std::vector::emplace_back()
Causes crashing in win32.

MartijnKaijser committed:
Merge pull request #10110 from MartijnKaijser/win32_installer

[win32] Improve installer and add appxmanifest

MartijnKaijser committed:
[win32] fix testbuilds

jenkins4kodi committed:
Merge pull request #10116 from MartijnKaijser/revert

MartijnKaijser committed:
Merge pull request #10117 from MartijnKaijser/win32_test

[win32] fix testbuilds

FernetMenta committed:
Merge pull request #10112 from FernetMenta/ratio

AE: fix resetting rr to 1.0
